blog/content/post/2021-10-10-nixos-with-heads.md

```diff
@@ -0,0 +1,548 @@
+	32768, 98304, 163840, 229376, 294912, 819200, 884736, 1605632, 2654208,
+	4096000, 7962624, 11239424, 20480000, 23887872
+
+Allocating group tables:    0/1274 done
+Writing inode tables:    0/1274 done
+Creating journal (262144 blocks): done
+Writing superblocks and filesystem accounting information:    0/1274 done
+```
+
+8. Create swap on the `swap` volume:
+
+```
+# mkswap -L swap /dev/vg/swap
+Setting up swapspace version 1, size = 8 GiB (8589930496 bytes)
+LABEL=swap, UUID=4c021582-f0a9-4348-9565-267fa8f3d1b1
+```
+
+### Proceeding with installation
+
+Once we have the partitioning in place, we can proceed with the installation.
+Before we start, we need to mount our partitions under `/mnt`:
+
+```bash
+# mount /dev/vg/root /mnt/
+# mkdir /mnt/boot
+# mount /dev/sda1 /mnt/boot
+# swapon /dev/vg/swap
+```
+
+The actual installation process consists of two steps. The first one is to edit
+the `/mnt/etc/nixos/configuration.nix` configuration file, which describes the
+whole content and configuration of our system.
+
+```bash
+# nixos-generate-config  --root /mnt/
+writing /mnt/etc/nixos/hardware-configuration.nix...
+
+# vi /mnt/etc/nixos/hardware-configuration.nix
+```
+
+In my case, I have enabled Wayland, sway, and some applications for my sway
+desktop environment:
+
+```
+  # Disable the X11 windowing system.
+  services.xserver.enable = false;
+
+  # Use Wayland with sway
+  programs.sway = {
+    enable = true;
+    wrapperFeatures.gtk = true; # so that gtk works properly
+    extraPackages = with pkgs; [
+      swaylock # screen locker
+      swayidle # idle management
+      xwayland
+      wl-clipboard
+      mako # notoficatoins
+      alacritty # terminal emulator
+      sway-contrib.grimshot # screenshot
+      wdisplays # displays configuration
+      wob # volume/backlight bar
+      wofi # launcher
+      waybar # top bar
+      oguri # wallpaper
+      autotiling
+      wlsunset
+      lxsession
+      udiskie
+      zathura # PDF viewer
+      imv # image viewer
+    ];
+  };
+```
+
+I also made some more configuration later on, but I can write more about that
+some other time. The only configuration I want to mention here (as it directly
+impacts the boot process) is the partitions configuration.
+
+The `/mnt/etc/nixos/hardware-configuration.nix` was generated automatically
+when running the `nixos-generate-config`, but it failed to correctly describe
+my logical volumes. I could not find a way how to properly fix that. I could
+not override the settings in the `configuration.nix`, as the `nixos-install`
+would complain:
+
+```bash
+building the configuration in /mnt/etc/nixos/configuration.nix...
+error The option `fileSystems./.device' has conflicting definition values:
+- In `/mnt/etc/nixos/configuration.nix': "/dev/mapper/vg-root"
+- In `/mnt/etc/nixos/hardware-configuration.nix': "/dev/disk/by-uuid/6839a958-2b1b-4548-909a-e479a02cb800"
+(use '--show-trace' to show detailed location information)
+```
```

Comment:
Maybe, I suspect, I could do something wrong. The installation manual does not cover the setup with encrypted disk. So maybe a chapter on this could be added.

Reply:
Yeah, I hope this post would not be toooooo long